skyportal/handlers/api/candidate/scan_report_item.py

```diff
@@ -32,12 +32,21 @@ def create_scan_report_item(session, report, sources_by_obj):
     obj = session.scalar(
         Obj.select(session.user_or_token, mode="read").where(Obj.id == obj_id)
     )
-    current_mag = obj.photstats[0].last_detected_mag if obj.photstats else None
-    current_age = (
-        (Time.now().mjd - obj.photstats[0].first_detected_mjd)
-        if obj.photstats
-        else None
-    )
+
+    if obj.photstats:
+        current_mag = obj.photstats[0].last_detected_mag
+        current_age = Time.now().mjd - obj.photstats[0].first_detected_mjd
+        if obj.redshift:
+            abs_mag_and_filter = {
+                "abs_mag": current_mag - obj.dm,
```

Comment:
you don't always have a dm value, for example we can't provide one if the redshift is too low. I would suggest just doing `dm = obj.dm if obj.redshift else None`, and then having your if statement on `if dm is not None:` or similar.